resources/src/main/java/org/robolectric/res/StringResources.java

```diff
@@ -96,4 +96,8 @@ static String escape(String text) {
       throw new IllegalArgumentException("Invalid code point: \\u" + codePoint, e);
     }
   }
+
+  private static String removeUnescapedDoubleQuotes(String input) {
+    return input.replaceAll("\"", "");
```

Comment:
I must have missed something, but I thought this is the approach I am using. 

I have switched away from using regex. Instead, I have built on top of the current char-by-char approach of parsing/scanning, and created `removeUnescapedDoubleQuotes()` method to do a second pass check on the substrings to strip away double quotes before appending to the final result.

I have added a test using the examples you listed above. Please let me know if I misunderstood your comment. Thank you!